From 515374d43148864fbdb27f9e26dac6679c0d93ed Mon Sep 17 00:00:00 2001
From: eric <eric.peterson1@twcable.com>
Date: Wed, 16 Sep 2015 16:19:07 -0600
Subject: [PATCH] Improving find  static robustness

The find static utility method assumes a filesystem layout
that may or may not be present on certain distros.  This fixes
this by using the python install path.

Closes-bug: #1494171

Change-Id: Iab72b4b872849d2496cc634677e5bfe478aa03be
---
 openstack_dashboard/settings.py        |  2 +-
 openstack_dashboard/static_settings.py | 11 ++++++++---
 openstack_dashboard/test/settings.py   |  2 +-
 3 files changed, 10 insertions(+), 5 deletions(-)

diff --git a/openstack_dashboard/settings.py b/openstack_dashboard/settings.py
index 1a05135..249e53a 100644
--- a/openstack_dashboard/settings.py
+++ b/openstack_dashboard/settings.py
@@ -325,7 +325,7 @@ if os.path.exists(os.path.join(CUSTOM_THEME, 'img')):
 
 # populate HORIZON_CONFIG with auto-discovered JavaScript sources, mock files,
 # specs files and external templates.
-find_static_files(ROOT_PATH, HORIZON_CONFIG)
+find_static_files(HORIZON_CONFIG)
 
 # Load the pluggable dashboard settings
 import openstack_dashboard.enabled
diff --git a/openstack_dashboard/static_settings.py b/openstack_dashboard/static_settings.py
index 4f0c4d3..1703680 100644
--- a/openstack_dashboard/static_settings.py
+++ b/openstack_dashboard/static_settings.py
@@ -139,12 +139,17 @@ def get_staticfiles_dirs(webroot='/'):
     return STATICFILES_DIRS
 
 
-def find_static_files(ROOT_PATH, HORIZON_CONFIG):
+def find_static_files(HORIZON_CONFIG):
+    import horizon
+    import openstack_dashboard
+    os_dashboard_home_dir = openstack_dashboard.__path__[0]
+    horizon_home_dir = horizon.__path__[0]
+
     # note the path must end in a '/' or the resultant file paths will have a
     # leading "/"
     file_discovery.populate_horizon_config(
         HORIZON_CONFIG,
-        os.path.join(ROOT_PATH, '..', 'horizon', 'static/')
+        os.path.join(horizon_home_dir, 'static/')
     )
 
     # filter out non-angular javascript code and lib
@@ -155,6 +160,6 @@ def find_static_files(ROOT_PATH, HORIZON_CONFIG):
     # leading "/"
     file_discovery.populate_horizon_config(
         HORIZON_CONFIG,
-        os.path.join(ROOT_PATH, 'static/'),
+        os.path.join(os_dashboard_home_dir, 'static/'),
         sub_path='app/'
     )
diff --git a/openstack_dashboard/test/settings.py b/openstack_dashboard/test/settings.py
index 1826f74..1926644 100644
--- a/openstack_dashboard/test/settings.py
+++ b/openstack_dashboard/test/settings.py
@@ -87,7 +87,7 @@ settings.update_dashboards(
 )
 INSTALLED_APPS[0:0] = []
 
-find_static_files(ROOT_PATH, HORIZON_CONFIG)
+find_static_files(HORIZON_CONFIG)
 
 # Set to True to allow users to upload images to glance via Horizon server.
 # When enabled, a file form field will appear on the create image form.
-- 
2.4.3

